gstreamer/src/segment_serde.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright (C) 2018 François Laignel <fengalin@free.fr>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use serde::de::{Deserialize, Deserializer};
+use serde::ser::{Serialize, SerializeStruct, Serializer};
+
+use Format;
+use GenericFormattedValue;
+use Segment;
+use SegmentFlags;
+
+impl<'a> Serialize for Segment {
```

Comment:
Does this also allow to work with `FormattedSegment`s directly, or would one have to cast first?

Reply:
I'm not quite sure. I must admit I don't have experience with the Formatted things so it's kind of fuzzy for me. Will investigate.

Reply:
Maybe there could be a simple macro implementation for those around the casts